prow/spyglass/gcsartifact_fetcher.go

```diff
@@ -142,6 +146,43 @@ func (af *GCSArtifactFetcher) artifacts(key string) ([]string, error) {
 	return artifacts, nil
 }
 
+func (af *GCSArtifactFetcher) signURL(bucket, obj string) (string, error) {
+	// If we're anonymous we can just return a plain URL.
+	if af.gcsCredsFile == "" {
+		artifactLink := &url.URL{
+			Scheme: httpsScheme,
+			Host:   "storage.googleapis.com",
+			Path:   path.Join(bucket, obj),
+		}
+		return artifactLink.String(), nil
+	}
+
+	// As far as I can tell, there is no sane way to get these values other than just
+	// reading them out of the JSON file ourselves.
+	f, err := os.Open(af.gcsCredsFile)
```

Comment:
Are we using the `SecretAgent`? Can we pass in the `func() []byte` accessor instead of the path?

Reply:
We are not.

Reply:
Let's go ahead and use it -- it's the canonical "reload this file for a secret value" thing everywhere else in Prow

Reply:
(feel free to hold cancel and address as a follow up)